File: C_ProblemSet1/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    //declaration of variables
    long cntr = 0;
    long rem = 0;
    long newrem = 0;
    long remofnewrem = 0;
    long sum1 = 0;
    long sum2 = 0;
    long checksum = 0;
    long counter = 0;
    //take input
    long card_number = get_long("Number: ");
    long temp = card_number;
    long temp1 = card_number;
    long temp2 = card_number;
    //checksum
    while (temp != 0)
    {
        temp = temp / 10;
        rem = temp % 10;
        newrem = rem * 2;
        while (newrem != 0)
        {
            remofnewrem = newrem % 10;
            sum1 = sum1 + remofnewrem;  // adding individual digits
            newrem = newrem / 10;
        }
        temp = temp / 10;
    }
    while (temp1 != 0)
    {
        rem = temp1 % 10;
        sum2 = sum2 + rem;
        temp1 = temp1 / 10;
        temp1 = temp1 / 10;
    }
    while (card_number != 0)
    {
        cntr++;
        card_number = card_number / 10;
    }
    counter = cntr;
    while (temp2 != 0)  //counter for checking starting digits
    {
        temp2 = temp2 / 10;
        if (counter == 3)   //starting 2 digits
        {
            rem = temp2;
        }
        else if (counter == 2) // starting 1 digit
        {
            newrem = temp2;
        }
        counter--;
    }
    checksum = sum1 + sum2;
    if (checksum % 10 == 0 && (cntr == 13 || cntr == 16) && newrem == 4)
    {
        printf("VISA\n");
    }
    else if (checksum % 10 == 0 && cntr == 16 && (rem == 51 || rem == 52 || rem == 53 || rem == 54 || rem == 55))
    {
        printf("MASTERCARD\n");
    }
    else if (checksum % 10 == 0 && cntr == 15 && (rem == 34 || rem == 37))
    {
        printf("AMEX\n");
    }
    else
    {
        printf("INVALID\n");
    }

}

File: Python_Lab_ProblemSet6/sentimental-credit/credit.py
from cs50 import get_string

global card_number


def main():
    # Ask for user for card
    card_number = get_string("Number: ")
    # ask for validating card
    if validate_card(card_number):
        # Identifies the company of the card
        company_card(card_number)
    else:
        print("INVALID\n")


def validate_card(card_number):
    sum_of_number_by_2 = 0
    temp_1 = int(card_number)
    # looping through second last digits to start alternately
    while temp_1 != 0:
        temp_1 = temp_1 // 10
        # multiplying it by 2
        rem = (temp_1 % 10)*2
        # performing addition on individual digits
        while rem != 0:
            new_rem = rem % 10
            sum_of_number_by_2 = sum_of_number_by_2 + new_rem
            rem = rem // 10
        temp_1 = temp_1 // 10

    alt_sum_of_number_by_2 = 0
    temp = int(card_number)
    # looping through last digits to start alternately
    while temp != 0:
        rem = temp % 10
        # adding sums of those digits
        alt_sum_of_number_by_2 = alt_sum_of_number_by_2+rem
        temp = temp // 10
        temp = temp // 10
    # addition of both sums
    final_sum = sum_of_number_by_2 + alt_sum_of_number_by_2
    if final_sum % 10 == 0:
        return True
    else:
        return False


def company_card(card_number):
    # checks if card is american express
    if card_number[0] == "3" or card_number[1] == "4" or card_number[1] == "7" and len(card_number) == 15:
        print("AMEX")
    # checks if card is mastercard
    elif card_number[0] == "5" and card_number[1] == "1" or card_number[1] == "2" or card_number[1] == "3" or card_number[1] == "4" or card_number[1] == "5" and len(card_number) == 16:
        print("MASTERCARD")
    # checks if card is visa
    elif card_number[0] == "4" and len(card_number) == 13 or len(card_number) == 16:
        print("VISA")
    else:
        print("INVALID\n")


# main function
main()
